core/domain/exp_domain.py

```diff
@@ -2817,9 +2842,22 @@ def _convert_v32_dict_to_v33_dict(cls, exploration_dict):
 
         return exploration_dict
 
+    @classmethod
+    def _convert_v33_dict_to_v34_dict(cls, exploration_dict, image_id_counter):
```

Comment:
Not sure if it is the best idea to pass image_id_counter into the function. I think you should compute the count of images and store it (/cc @DubeySandeep for his opinion here).

Reply:
DONE